pkg/gatherers/common/gather_logs.go

```diff
@@ -156,38 +178,35 @@ func filterLogs(
 	}()
 
 	scanner := bufio.NewScanner(stream)
-	return FilterLogFromScanner(scanner, messagesToSearch, regexSearch, nil)
+	return FilterLogFromScanner(scanner, messagesToSearch, messagesRegexp, nil)
 }
 
 // FilterLogFromScanner filters the desired messages from the log
-func FilterLogFromScanner(scanner *bufio.Scanner, messagesToSearch []string, regexSearch bool,
+func FilterLogFromScanner(scanner *bufio.Scanner, messagesToSearch []string, messagesRegexp *regexp.Regexp,
 	cb func(lines []string) []string) (string, error) {
 	var result []string
 
-	var messagesRegexp *regexp.Regexp
-	if regexSearch {
-		messagesRegexp = regexp.MustCompile(strings.Join(messagesToSearch, "|"))
-	}
-
 	for scanner.Scan() {
 		line := scanner.Text()
-		if len(messagesToSearch) == 0 {
-			result = append(result, line)
-			continue
-		}
 
-		if regexSearch && messagesRegexp != nil {
+		if messagesRegexp != nil {
 			matches := messagesRegexp.MatchString(line)
 			if matches {
 				result = append(result, line)
 			}
-		} else {
-			for _, messageToSearch := range messagesToSearch {
-				if strings.Contains(strings.ToLower(line), strings.ToLower(messageToSearch)) {
-					result = append(result, line)
-				}
+			continue
```

Comment:
So when there's some `messageRegexp` provided then the `messagesToSearch` are not taken into account at all right? If so then I would probably split it in two functions or at least provide some better documentation for the function.

Reply:
@tremes I extended the log functions a little bit now and also added some comments. Please, check it. I think it looks nice...